plugins/auto-sizes/tests/e2e/specs/improve-calculate-sizes.spec.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+
+/**
+ * WordPress dependencies
+ */
+const { test } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'check accurate sizes', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyfour' );
+		await requestUtils.deactivatePlugin( 'enhanced-responsive-images' );
+		await requestUtils.deleteAllMedia();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllMedia();
+	} );
+
+	test( 'should get smaller version of image', async ( {
+		admin,
+		editor,
+		requestUtils,
+		page,
+	} ) => {
+		const filename = 'leaves.jpg';
+		const filepath = path.join(
+			__dirname + '/../../data/images/',
+			filename
+		);
+
+		await admin.createNewPost();
+		const media = await requestUtils.uploadMedia( filepath );
+
+		await editor.insertBlock( {
+			name: 'core/group',
+			attributes: {
+				align: 'wide',
+			},
+			innerBlocks: [
+				{
+					name: 'core/image',
+					attributes: {
+						alt: filename,
+						id: media.id,
+						url: media.source_url,
+					},
+				},
+			],
+		} );
+
+		const postId = await editor.publishPost();
+
+		// Navigate to the post and wait for the image to load.
+		await page.goto( `/?p=${ postId }` );
+		const imageElement = await page.waitForSelector(
+			`img.wp-image-${ media.id }`
+		);
+		const imageSizes = await imageElement.getAttribute( 'sizes' );
+
+		// Activate the plugin.
+		await requestUtils.activatePlugin( 'enhanced-responsive-images' );
+
+		// Reload the page and wait for the image to load.
+		await page.goto( `/?p=${ postId }` );
+		const updatedImageElement = await page.waitForSelector(
+			`img.wp-image-${ media.id }`
+		);
+
+		const updatedImageSizes =
+			await updatedImageElement.getAttribute( 'sizes' );
+
+		if ( imageSizes === updatedImageSizes ) {
+			throw new Error(
+				'Image sizes did not update after activating the plugin.'
+			);
+		}
+
+		if ( '(max-width: 620px) 100vw, 620px' !== updatedImageSizes ) {
+			throw new Error(
+				`Unexpected image sizes: ${ updatedImageSizes }. Expected: (max-width: 620px) 100vw, 620px`
+			);
+		}
+
+		const currentSrc = await updatedImageElement.evaluate( ( img ) =>
+			img instanceof HTMLImageElement ? img.currentSrc : null
+		);
+		currentSrc.endsWith( 'leaves-768x512.jpg' );
```

Comment:
This doesn't do anything, this isn't an assertion that fails the test.

Ideally we use something like `await expect(locator).toHaveAttribute('src', /leaves-768x512\.jpeg$/ );`

Reply:
Addressed in https://github.com/WordPress/performance/pull/1988/commits/004aa65d5b9af26ac9073fdb180eb8c66ff8cb0b